### PR TITLE
skip scanallip if input is ip

### LINKED ISSUE #3185: scanAllIps: failed to parse hostname when using uncover query
### Nuclei version:
latest,dev

### Current Behavior:
```console
nuclei -uq 'ssl:"uber.com"' -id tech-detect -debug

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.6

		projectdiscovery.io

[INF] Running uncover query against: shodan
[DBG] scanAllIps: failed to parse hostname of 104.17.12.50:2087 falling back to default
[DBG] scanAllIps: failed to parse hostname of 23.8.29.137:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 23.192.28.12:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 23.78.105.200:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 104.87.106.46:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 104.96.141.115:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 2.17.56.70:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 157.197.177.121:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 34.98.127.226:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 34.254.86.27:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 223.119.227.29:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 23.222.158.219:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 184.50.221.34:443 falling back to default
[DBG] scanAllIps: failed to parse hostname of 23.198.82.202:443 falling back to default
```

### Expected Behavior:
running uncover query without error/issue

### Steps To Reproduce:
```
nuclei -uq 'ssl:"uber.com"' -id tech-detect -debug
```

### PR DESCRIPTION
**Pull request by tarunKoyalwar**
## Proposed changes

- Adds Condition check and scanallips is skipped if given input is ip 

closes #3185 

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tarunKoyalwar** [comment]
# Changes

- removes `defer ratelimit.Stop()`  which caused infinite wait 
- adds `urlutil.ParseWithSchema` to parse URL

**Comment by tarunKoyalwar** [comment]
```sh
$ ./nuclei -uc  -uq 'ssl:"uber.com"' -id tech-detect   

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.6

		projectdiscovery.io

[INF] Running uncover query against: shodan
[INF] Using Nuclei Engine 2.8.6 (latest)
[INF] Using Nuclei Templates 9.3.4 (latest)
[INF] Templates added in last update: 15
[INF] Templates loaded for scan: 1
[INF] Targets loaded for scan: 200
[INF] Running httpx on input host
[INF] Found 200 URL from httpx
[tech-detect:cloudflare] [http] [info] https://104.17.141.8:2087
[tech-detect:cloudflare] [http] [info] https://104.17.177.5:8443
[tech-detect:cloudflare] [http] [info] https://104.17.177.5:2083
[tech-detect:cloudflare] [http] [info] https://104.16.55.17:443
[tech-detect:cloudflare] [http] [info] https://104.17.141.8:2083
[tech-detect:cloudflare] [http] [info] https://104.18.252.218:2083
[tech-detect:cloudflare] [http] [info] https://104.16.55.17:8443
[tech-detect:cloudflare] [http] [info] https://104.18.253.218:443
[tech-detect:cloudflare] [http] [info] https://104.16.54.17:443
[tech-detect:cloudflare] [http] [info] https://104.17.12.50:2087
[tech-detect:cloudflare] [http] [info] https://104.17.178.5:443
[tech-detect:akamai] [http] [info] https://104.114.155.39:443
[tech-detect:akamai] [http] [info] https://104.120.153.94:443
[tech-detect:akamai] [http] [info] https://104.71.75.58:443
[tech-detect:akamai] [http] [info] https://104.67.175.201:443
[tech-detect:akamai] [http] [info] https://104.69.182.78:443
[tech-detect:akamai] [http] [info] https://104.119.150.236:443
[tech-detect:akamai] [http] [info] https://104.66.189.165:443
....
```

**Comment by tarunKoyalwar** [comment]
gh actions integration test seems like false positive works locally !!

File: v2/pkg/core/inputs/hybrid/hmap.go [hunk 1]
```diff
@@ -171,24 +171,30 @@ func (i *Input) Set(value string) {
 		return
 	}
 	// parse hostname if url is given
-	host := utils.ParseHostname(value)
-	if host == "" {
-		// not a valid url hence scanallips is skipped
-		gologger.Debug().Msgf("scanAllIps: failed to parse hostname of %v falling back to default", value)
-		i.setItem(&contextargs.MetaInput{Input: value})
+	urlx, er := urlutil.ParseWithScheme(URL)
```
Comment (Mzack9999) [inline v2/pkg/core/inputs/hybrid/hmap.go:174]: what about `err`?

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

File: v2/pkg/utils/utils_test.go [hunk 2]
```diff
@@ -19,24 +19,3 @@ func TestUnwrapError(t *testing.T) {
 	errThree := fmt.Errorf("error with error: %w", errTwo)
 	require.Equal(t, errOne, UnwrapError(errThree))
 }
-
-func TestParseURL(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (tarunKoyalwar) [reply v2/pkg/utils/utils_test.go:23]: @Mzack9999  , we already have unit tests for this parsing issue at https://github.com/projectdiscovery/utils/blob/main/url/urlutil_test.go

**Review by Mzack9999 — APPROVED** [review]

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**